### DISCUSSION TITLE
embarrassments/

### DISCUSSION THREAD

**Discussion by giscus** [zmike/zmike.github.io#4] [Announcements]
# embarrassments/

This Week

https://www.supergoodcode.com/embarrassments/

**Comment by mbriar** [comment]
Wow, very confidence inspiring that the VK_EXT_vertex_input_dynamic_state implementation is robust when they block the only major user of it..

Also now zink not only destroys radeonsi in DOOM 2016, but also Iris  😂

**Comment by wertig0n** [comment]
Don't tell me we have to go back to User Agent Spoofing again, that was HELL! 😨

**Comment by atlasvirtualreality** [comment]
oh my oh my, we have to call these guys again and give them a piece of that fruit punch cake.

**Comment by mrdeathjr28** [comment]
Thanks for your work with my intel uhd 630 runs many games native and non native (old titles, i dont like newer mainly) curiously more smootly (much less stutter) compared with my before geforce gtx 1050 without forget runs some titles dont run on nvidia case pacman world 3 and others

Respect zink runs various emulators (case hades - nanoboy advance and maybe others) without vulkan renderer however stay waiting for add support to VK_EXT_attachment_feedback_loop_layout

However them stay working on that

https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/17892

In other native linux gaming test some titles case saint row the third, neverwinter nights EE, spec ops the line, open nox and others and runs very well

However some native titles dont run in my case like deadfall adventures, shadow warrior and shadow blade for now

**Reply by zmike** [reply]
Please file tickets if you find issues with specific games, otherwise I have no idea that they don't work.

**Reply by mrdeathjr28** [reply]
Yeah but i wait for VK_EXT_attachment_feedback_loop_layout stay merged on mesa

Hopefully Lionel Landwerlin stay working on that and Faith Ekstrand

**Comment by rijnhard** [comment]
Well _putting trust in the gap_ I'd lean towards AMD tested it with zink and it broke something, hence they blocked it. Hopefully they will unblock it soonish though.

**Comment by vsyrjala** [comment]
> The perf on zink is so good that the game thinks it’s running on an NVIDIA GPU.

Are we sure the performance being so good  isn't *because* the game thinks it's running on a NVIDIA GPU?

**Reply by mbriar** [reply]
Yeah, i'm pretty sure about that. Radeonsi is also really slow for some reason in DOOM 2016 and spoofing nvidia doesn't change anything.

**Reply by zmike** [reply]
All I know is I didn't change anything in my environment and somehow the game thinks this is an nvidia gpu.

**Reply by atlasvirtualreality** [reply]
do you plan on including it in proton for the future if the performance surpasses the native opengl implementation, also can't developer fix any bug that they have in their game instead of you guys making custom for each of them which is time consuming, i understand if the fixes are temporary but dev need to take the responsibility and fix their bugs.

**Reply by zmike** [reply]
I don't think proton is going to start including GPU drivers any time soon, so probably not.

**Reply by jntesteves** [reply]
> I don't think proton is going to start including GPU drivers any time soon, so probably not.

* DXVK 🫥
* VKD3D-proton 😶‍🌫️

**Reply by mbriar** [reply]
Either way there's no reason to include it in proton because everyone has zink as part of their distro's mesa package already anyways.

**Reply by jntesteves** [reply]
> Either way there's no reason to include it in proton because everyone has zink as part of their distro's mesa package already anyways.

From the post:

> ...so long as you’re on Mesa 23.1-rc2 or a recent git build, I doubt you’ll notice the difference for most games.

**No one** has that as part of their distro's mesa package. Many don't even have a working version of zink. What you said is simply not true.

**Reply by zmike** [reply]
> > I don't think proton is going to start including GPU drivers any time soon, so probably not.
> 
> * DXVK 🫥
> * VKD3D-proton face_in_clouds

Those aren't drivers, they're layering libraries. It's a subtle but distinct difference.

**Reply by jntesteves** [reply]
> > * DXVK 🫥
> > * VKD3D-proton face_in_clouds
> 
> Those aren't drivers, they're layering libraries. It's a subtle but distinct difference.

I thought so too. But then, in this context, what makes zink a driver?

**Reply by zmike** [reply]
Being a driver makes it a driver.

**Reply by jntesteves** [reply]
TIL: never question Big Triangle

**Reply by atlasvirtualreality** [reply]
Mike sorry I didn't it was automatically enabled in the mean driver stack that's why I asked if it was gonna be in proton also will you active it for steamos soon to run openly games under zink or your gonna work on it further?

**Reply by atlasvirtualreality** [reply]
Sorry for typos I wa writing on my phone.

**Reply by zmike** [reply]
I don't make the decisions about what steam uses, but I imagine if there's cases where zink consistently outperforms native drivers then it will be enabled there.

**Reply by atlasvirtualreality** [reply]
Mike you're my hero.

**Comment by marcosps** [comment]
Hey! Thanks a lot for suggesting giscus! I started using in my hugo blog, and it's really amazing!

Also, very nice post as usual!
